src/services/side-panels-manager/models/item-wrapper.tsx

```diff
@@ -12,43 +11,55 @@ import { PanelComponentProps, SidePanelItem } from './side-panel-item';
 export class ItemWrapper {
   private static nextId = 0;
   public readonly id: number;
+  public iconId: number | undefined;
   public readonly item: SidePanelItem;
   private readonly player: KalturaPlayer;
+  private readonly upperBarManager: UpperBarManager;
   private panelItemComponentRef!: RefObject<PanelItemWrapper>;
   private removePanelComponentFn!: () => void;
-  private iconComponentRef: RefObject<IconWrapper> | undefined;
-  private removeIconComponentFn: (() => void) | undefined;
   private isActive: boolean;
   constructor(item: SidePanelItem, player: KalturaPlayer, onClick: (panelItemId: number) => void) {
     this.id = ++ItemWrapper.nextId;
     this.item = item;
     this.player = player;
+    this.upperBarManager = this.player.getService<UpperBarManager>('upperBarManager');
     this.isActive = false;
     this.injectPanelComponent();
-    if (item.iconComponent) this.injectIconComponent(onClick);
+    if (item.iconComponent) {
```

Comment:
is it possible for IconComponent (or something like "IconWrapper") to extend ItemWrapper instead of having logic specific to icon at the item wrapper level ?